src/main/scala/io/iohk/ethereum/network/discovery/PeerDiscoveryManager.scala

```diff
@@ -42,13 +45,21 @@ class PeerDiscoveryManager(
   }
 
   def scan(): Unit = {
-    // Always ping bootstrap nodes as they have many neighbours and we do not want lose any info about them
-    // One optimisation worth considering is pinging them only every n-th scan
-    bootStrapNodesInfo.foreach{ nodeInfo =>
+    // At the beginning ping all bootstrap nodes
+    if (firstScan) {
+      bootStrapNodesInfo.foreach{ nodeInfo =>
+        sendPing(Endpoint.makeEndpoint(nodeInfo.node.udpSocketAddress, nodeInfo.node.tcpPort), nodeInfo.node.udpSocketAddress, nodeInfo)
+      }
+
+      firstScan = false
+    }
+
+    // Ping a random sample from currently pinged nodes without the answer
```

Comment:
There was a comment on the previous version about 

> One optimisation worth considering is pinging them only every n-th scan

The goal of this is the same, right?

Reply:
Yes, it is one of its goals but this approach is more robust imo.
At the beginning of syncing, the sample will be composed only from our bootstrap nodes so unless we would get more new node we will constantly re-ping bootstraps.
And later, It can help us with the case that perfectly good peer pong will be missed because of udp having no guarantees of delivery. (we will constantly re-ping a sample from unanswered ping)
This approach is not perfect but should help us with cases, that because of missed some valid pongs we will be stuck. (now it can happen from time to time and one solution is to restart mantis with hope that this time `pongs` will get through)